core/server/api_container/server/startosis_engine/startosis_validator/validator_environment.go

```diff
@@ -14,6 +15,7 @@ import (
 type ValidatorEnvironment struct {
 	imagesToPull                  map[string]bool // "set" of images that need to be downloaded
 	imagesToBuild                 map[string]*image_build_spec.ImageBuildSpec
+	imagesToPullWithAuth          map[string]*image_registry_spec.ImageRegistrySpec
```

Comment:
wonder if it's possible to merge `imagesToPull` and `imagesToPullWithAuth` into one `imagesToPull map[string]*image_registry_spec.ImageRegistrySpec`, where raw container image strings are just `ImageRegistrySpec` with no auth/default registry?

Reply:
Yep I thought of this as well but was worried about introducing nulls, though it should be okay.